data-prepper-plugins/kinesis-source/src/main/java/org/opensearch/dataprepper/plugins/kinesis/source/KinesisMultiStreamTracker.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ */
+
+package org.opensearch.dataprepper.plugins.kinesis.source;
+
+import org.opensearch.dataprepper.plugins.kinesis.source.configuration.KinesisSourceConfig;
+import org.opensearch.dataprepper.plugins.kinesis.source.configuration.KinesisStreamConfig;
+import software.amazon.awssdk.services.kinesis.KinesisAsyncClient;
+import software.amazon.awssdk.services.kinesis.model.DescribeStreamRequest;
+import software.amazon.awssdk.services.kinesis.model.DescribeStreamResponse;
+import software.amazon.awssdk.services.kinesis.model.StreamDescription;
+import software.amazon.kinesis.common.InitialPositionInStreamExtended;
+import software.amazon.kinesis.common.StreamConfig;
+import software.amazon.kinesis.common.StreamIdentifier;
+import software.amazon.kinesis.processor.FormerStreamsLeasesDeletionStrategy;
+import software.amazon.kinesis.processor.MultiStreamTracker;
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+
+
+public class KinesisMultiStreamTracker implements MultiStreamTracker {
+    private static final String COLON = ":";
+
+    private final KinesisAsyncClient kinesisClient;
+    private final KinesisSourceConfig sourceConfig;
+    private final String applicationName;
+
+    public KinesisMultiStreamTracker(KinesisAsyncClient kinesisClient, final KinesisSourceConfig sourceConfig, final String applicationName) {
+        this.kinesisClient = kinesisClient;
+        this.sourceConfig = sourceConfig;
+        this.applicationName = applicationName;
+    }
+
+    @Override
+    public List<StreamConfig> streamConfigList()  {
+        List<StreamConfig> streamConfigList = new ArrayList<>();
+        for (KinesisStreamConfig kinesisStreamConfig : sourceConfig.getStreams()) {
+            StreamConfig streamConfig;
+            try {
+                streamConfig = getStreamConfig(kinesisStreamConfig);
+            } catch (Exception e) {
+                throw new RuntimeException(e);
```

Comment:
What kind of exceptions get thrown here? If it's a config error we should throw InvalidPluginConfigurationException and provide context with the message where we can

Reply:
This is an internal `StreamConfig` class which is passed to KCL. It is not data prepper kinesis stream config.